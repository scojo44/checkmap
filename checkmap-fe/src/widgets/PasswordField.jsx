import React from 'react'
import FormField from './FormField'

export default function PasswordField({name, label, inputAttributes = {}, register, validation, errors}) {
  return (
    <FormField inputType="password" inputAttributes={{...inputAttributes}}
      {...{name, label, register, errors, validation}}
    />
  );
}
