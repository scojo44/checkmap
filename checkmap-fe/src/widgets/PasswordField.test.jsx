import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import PasswordField from './PasswordField'

const fakeRegister = vi.fn((name, options) => {
  return {
    name,
    onChange: x => x,
    onBlur: x => x
  }
});

const fakeErrors = {
  test: {message: 'test message'}
}

describe('PasswordField Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <PasswordField name="password" label="Password" register={fakeRegister} errors={fakeErrors}/>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <PasswordField name="password" label="Password" register={fakeRegister} errors={fakeErrors}/>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
