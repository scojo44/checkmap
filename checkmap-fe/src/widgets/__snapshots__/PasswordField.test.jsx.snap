// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PasswordField Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <p
    class="FormField"
  >
    <label
      for="password"
    >
      Password: 
    </label>
    <input
      name="password"
      type="password"
    />
  </p>
</DocumentFragment>
`;
